libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -534,6 +545,13 @@ static void eventCallback( MQTTContext_t * pContext,
             incomingInfo.pPayload = malloc( pPublishInfo->payloadLength );
             TEST_ASSERT_NOT_NULL( incomingInfo.pPayload );
             memcpy( ( void * ) incomingInfo.pPayload, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+
+            /* Update the global variable if the incoming PUBLISH packet
+             * represents a retained message. */
+            if( pPublishInfo->retain == true )
+            {
+                receivedRetainedMessage = true;
+            }
```

Comment:
If `receivedRetainedMessage` is not expected to keep its value across publishes, then this could just be:
```suggestion
            receivedRetainedMessage = pPublishInfo->retain;
```